cmd/lifecycle-util.go

```diff
@@ -0,0 +1,277 @@
+	rightAlign  int = 3
+)
+
+const (
+	idLabel             string = "ID"
+	prefixLabel         string = "Prefix"
+	statusLabel         string = "Enabled "
+	statusDisabledLabel string = "Disabled"
+	expiryLabel         string = "Expiry"
+	expiryDatesLabel    string = "Date/Days "
+	singleTagLabel      string = "Tag"
+	tagLabel            string = "Tags"
+	transitionLabel     string = "Transition"
+	transitionDateLabel string = "Date/Days "
+	storageClassLabel   string = "Storage-Class "
+	forceLabel          string = "force"
+	allLabel            string = "all"
+)
+
+const (
+	statusLabelKey          string = "Enabled"
+	storageClassLabelKey    string = "Storage-Class"
+	expiryDatesLabelKey     string = "Expiry-Dates"
+	transitionDatesLabelKey string = "Transition-Date"
+	transitionDaysLabelKey  string = "Transition-Days"
+)
+
+const (
+	expiryDatesLabelFlag string = "Expiry-Date"
+	expiryDaysLabelFlag  string = "Expiry-Days"
+)
+
+const (
+	fieldMainHeader         string = "Main-Heading"
+	fieldThemeHeader        string = "Row-Header"
+	fieldThemeRow           string = "Row-Normal"
+	fieldThemeTick          string = "Row-Tick"
+	fieldThemeExpiry        string = "Row-Expiry"
+	fieldThemeResultSuccess string = "SucessOp"
+	fieldThemeResultFailure string = "FailureOp"
+)
+
+const (
+	tickCell      string = "\u2713 "
+	crossTickCell string = "\u2717 "
+	blankCell     string = "  "
+)
+
+const defaultILMDateFormat string = "2006-01-02"
+
+const (
+	tagSeperator    string = ","
+	keyValSeperator string = ":"
+	tableSeperator  string = "|"
+)
+
+type tableCellInfo struct {
+	label       string
+	multLabels  []string
+	labelKey    string
+	fieldTheme  string
+	columnWidth int
+	align       int
+}
+
+type showDetails struct {
+	allAvailable bool
+	expiry       bool
+	transition   bool
+	json         bool
+	minimum      bool
+}
+
+func getCentered(label string, maxLen int) string {
+	const toPadWith string = " "
+	lblLth := len(label)
+	if lblLth > 1 && lblLth%2 != 0 {
+		lblLth++
+	} else if lblLth == 1 {
+		lblLth = 2
+	}
+	length := (float64(maxLen - lblLth)) / float64(2)
+	rptLth := (int)(math.Floor(length / float64(len(toPadWith))))
+	leftRptLth := rptLth
+	rightRptLth := rptLth
+	if rptLth <= 0 {
+		leftRptLth = 1
+		rightRptLth = 0
+	}
+	output := strings.Repeat(toPadWith, leftRptLth) + label + strings.Repeat(toPadWith, rightRptLth)
+	return output
+}
+
+func getLeftAlgined(label string, maxLen int) string {
+	const toPadWith string = " "
+	lblLth := len(label)
+	length := maxLen - lblLth
+	if length <= 0 {
+		return label
+	}
+	output := strings.Repeat(toPadWith, 1) + label + strings.Repeat(toPadWith, length-1)
+	return output
+}
+
+func getRightAligned(label string, maxLen int) string {
+	const toPadWith string = " "
+	lblLth := len(label)
+	length := maxLen - lblLth
+	if length <= 0 {
+		return label
+	}
+	output := strings.Repeat(toPadWith, length) + label
+	return output
+}
+
+func getBucketNameFromURL(urlStr string) string {
```

Comment:
url2BucketAndObject() is implemented for this reason alone - if client-s3.go is used then this should work appropriately.